### PR TITLE
C-1851 Implement Share to TikTok

### PR DESCRIPTION
**Pull request by nicoback2**
### Description
Implement Share to TikTok on iOS and Android
TODO - deal with errors handed back from TikTok

Lol sorry for these janky vids

https://user-images.githubusercontent.com/36916764/214170573-ae2cc6aa-5c2f-4519-b79d-cb28ad91a263.MOV

https://user-images.githubusercontent.com/36916764/214168483-89505473-5cd2-4c3d-8401-a19d8384dc24.MOV

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/src/components/share-drawer/ShareDrawer.tsx [hunk 1]
```diff
@@ -75,20 +74,20 @@ export const ShareDrawer = () => {
   const styles = useStyles()
   const viewShotRef = useRef() as React.RefObject<ViewShot>
 
-  const { isEnabled: isShareToTikTokEnabled } = useFeatureFlag(
+  const { isEnabled: isShareSoundToTikTokEnabled } = useFeatureFlag(
```
Comment (sliptype) [inline packages/mobile/src/components/share-drawer/ShareDrawer.tsx:77]: This is probably never turning back on :( but thanks for handling!

File: packages/mobile/src/components/share-drawer/useShareToStory.tsx [hunk 2]
```diff
@@ -354,12 +389,15 @@ export const useShareToStory = ({
...
             await pasteToInstagramApp(videoUri, stickerUri)
-          } else {
+          } else if (platform === 'snapchat') {
             await pasteToSnapchatApp(
               videoUri,
               `file://${stickerUri}`,
               content.track.permalink
             )
+          } else if (platform === 'tiktok') {
+            console.log(storyVideoPath, 'PATH')
```
Comment (sliptype) [inline packages/mobile/src/components/share-drawer/useShareToStory.tsx:399]: log

File: packages/mobile/src/components/share-drawer/useShareToStory.tsx [hunk 3]
```diff
@@ -320,9 +351,13 @@ export const useShareToStory = ({
...
         const SHOWFREQS_SEGMENT =
           'aformat=channel_layouts=mono:sample_rates=16000,adynamicsmooth,showfreqs=s=900x40:fscale=log:colors=#ffffff70'
+        const thirdLayerSegment =
+          platform === 'tiktok'
+            ? '[vid];[1:v]scale=396:548[stkr];[vid][stkr]overlay=format=auto:x=72:y=200;'
+            : ';'
         try {
           session = await FFmpegKit.execute(
-            `${audioStartOffsetConfig}-i ${streamMp3Url} -filter_complex "${backgroundSegment}[bg];[0:a]${SHOWFREQS_SEGMENT}[fg];[0:a]${SHOWFREQS_SEGMENT},vflip[fgflip];[bg][fg]overlay=format=auto:x=-100:y=H-h-100[fo];[fo][fgflip]overlay=format=auto:x=-100:y=H-h-60;[0:a]anull" -pix_fmt yuv420p -c:v libx264 -preset ultrafast -c:a aac -t 10 ${storyVideoPath}`
+            `${audioStartOffsetConfig}-i ${streamMp3Url} -i ${stickerUri} -filter_complex "${backgroundSegment}[bg];[0:a]${SHOWFREQS_SEGMENT}[fg];[0:a]${SHOWFREQS_SEGMENT},vflip[fgflip];[bg][fg]overlay=format=auto:x=-100:y=H-h-100[fo];[fo][fgflip]overlay=format=auto:x=-100:y=H-h-60${thirdLayerSegment}[0:a]anull" -pix_fmt yuv420p -c:v libx264 -preset ultrafast -c:a aac -t 10 ${storyVideoPath}`
```
Comment (sliptype) [inline packages/mobile/src/components/share-drawer/useShareToStory.tsx:360]: Should this stickerUri always be in the ffmpeg command? Or only for the tiktok case?

**Review by sliptype — APPROVED** [review]
Wow, this is clean. I like the refactors!

File: packages/mobile/src/components/share-drawer/useShareToStory.tsx (hunk 3 above)
Comment (nicoback2) [reply packages/mobile/src/components/share-drawer/useShareToStory.tsx:387]: ooh good catch, just for tiktok

**Pull request merged**